src/main/java/com/imsweb/datagenerator/utils/DistributionUtils.java

```diff
@@ -179,4 +182,20 @@ public static String getState() {
             _DIST_STATE = Distribution.of(_STATES);
         return _DIST_STATE.getValue();
     }
+
+
+    public static int getAgeGroup(String site) {
+        if ((site == null) || (Thread.currentThread().getContextClassLoader().getResource("frequencies/sites/" + site.toUpperCase() + ".csv") == null))
```

Comment:
It's good practice and safer to check that the resource exists, but that should be done only when you actually needs to load the resource. As it is now, you will check for the frequencies files every time a new frequency is requested. When generating a large number of records, you will need to load the file once, but you will request a frequency many time. Most of the time, you will find one in the static map and this method will return. You should check for the existence of the resource only if the "distribution" variable is null. That will be more efficient, and it makes sense logically.

Reply:
Fixed.